app/models/gallery.rb

```diff
@@ -29,7 +29,12 @@ class Gallery < ApplicationRecord
     # fetches an array of
     # galleries.map(&:gallery_groups).map{|group| [f1: group.id, f2: group.name]}
     # ordered by tag name
-    select("ARRAY(SELECT row_to_json(ROW(tags.id, tags.name)) FROM tags LEFT JOIN gallery_tags ON gallery_tags.tag_id = tags.id WHERE gallery_tags.gallery_id = galleries.id AND tags.type = 'GalleryGroup' ORDER BY gallery_tags.id ASC) AS gallery_groups_data_internal")
+    select("ARRAY(\
```

Comment:
Could we use HEREDOCs here? Especially squiggly ones, which will remove additional indentation for us!

Something like such (though I'm not sure about the position of the final "SQL", maybe it looks better elsewhere, but it seems like we can't put the end paren right after it):

```rb
  select(<<~SQL
    ARRAY(
      SELECT row_to_json(ROW(tags.id, tags.name)) FROM tags
      LEFT JOIN gallery_tags ON gallery_tags.tag_id = tags.id
      WHERE gallery_tags.gallery_id = galleries.id AND tags.type = 'GalleryGroup'
      ORDER BY gallery_tags.id ASC
    ) AS gallery_groups_data_internal
    SQL
  )
```

This should work, I think (but maybe test it? I'm not 100% sure!).

Reply:
There are some other `select`s that could do with this, too!

Reply:
❤️